libglusterfs/src/inode.c

```diff
@@ -449,13 +449,18 @@ __inode_retire(inode_t *inode)
 static int
 __inode_get_xl_index(inode_t *inode, xlator_t *xlator)
 {
-    int set_idx = -1;
+    int set_idx = xlator->level;
+
+    if (set_idx > inode->table->root_level)
+        set_idx = (inode->table->root_level + xlator->xl_id -
+                   inode->table->root_id);
```

Comment:
It's better to create a function to compute the index since it's used in several places.

Reply:
done